### ISSUE TITLE
Tag Helper is generating the empty link

### ISSUE THREAD

**Issue by Sourabh721992** [dotnet/core#6304]
I am trying creating link using tag helper but that is creating the empty link. I have included Tag helper reference as well. PFB the code:

@addTagHelper *,Microsoft.AspNetCore.Mvc.TagHelpers
<a asp-controller="home" asp-action="details" asp-route-id="@emp.Id" class="btn btn-primary">View</a>

Output 
<a class="btn btn-primary" href="">View</a>

**Comment by ManickaP** [comment]
Hi @Sourabh721992 could you please specify in more detail what you're doing and what you're trying to achieve? Could you share a minimal reproduction (source code) of the problem so we can help.

**Comment by Sourabh721992** [comment]
Hi,

I am using a tag anchor link to go from view, Index.cshtml to view, Details.cshtml. Both views are linked to HomeController in the solution. But what is happening is tag helper is creating empty link for me. I am attaching all the files.


[Index.txt](https://github.com/dotnet/core/files/6551521/Index.txt)
[Details.txt](https://github.com/dotnet/core/files/6551522/Details.txt)
[_ViewImports.txt](https://github.com/dotnet/core/files/6551524/_ViewImports.txt)
[HomeController.txt](https://github.com/dotnet/core/files/6551526/HomeController.txt)

**Comment by ManickaP** [comment]
Are you following any example or documentation?
https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/intro?view=aspnetcore-5.0 might help you, there are also some samples here: https://github.com/dpaquette/TagHelperSamples

Unfortunately, this is beyond my knowledge. @Pilchie is this something you can help with?

**Comment by Sourabh721992** [comment]
Yes I have followed the documentation but I am not sure why the link is href is not getting generated.
